lib/perl/Genome/Disk/Detail/Allocation/Creator.pm

```diff
@@ -29,7 +32,9 @@ sub create_allocation {
     my $allocation_object = $self->_get_allocation_without_lock(
         \@candidate_volumes);
 
-    $self->create_directory_or_delete_allocation($allocation_object);
+    if ($ENV{UR_DBI_NO_COMMIT} or not $skip_allocation_path_creation) {
```

Comment:
I kind of think this should just be
```perl
unless ($skip_allocation_path_creation) {
```
There's no particular reason why allocations themselves can't support skipping path creation under no-commit.  So I'd suggest we move this logic out and handle it in the cases that care.

As an added bonus, we could add a testcase to the `.t` that makes sure we don't create the path when the flag is enabled!